compiler/test/compilable/test16495.d

```diff
@@ -0,0 +1,22 @@
+/* TEST_OUTPUT:
+---
+trait   : test16495.C
+trait   : test16495.C[]
+trait   : const(test16495.C[])
+---
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=16495
+
+//import std.traits;
+
+class C { }
+
+//pragma(msg, "template: ", std.traits.fullyQualifiedName!C);
+pragma(msg, "trait   : ", __traits(fullyQualifiedName, C));
+
+//pragma(msg, "template: ", std.traits.fullyQualifiedName!(C[]));
+pragma(msg, "trait   : ", __traits(fullyQualifiedName, C[]));
+
+//pragma(msg, "template: ", std.traits.fullyQualifiedName!(const C[]));
+pragma(msg, "trait   : ", __traits(fullyQualifiedName, const C[]));
```

Comment:
The testsuite is already pretty chatty, which makes it difficult to use these within dejagnu, static asserts are preferred.